pkg/provision/providers/qemu/arch.go

```diff
@@ -200,11 +200,14 @@ func generateUEFIPFlashList(uefiSourcePathPrefixes, uefiSourceFiles, uefiVarsFil
 }
 
 // QemuExecutable returns name of qemu executable for the arch.
-func (arch Arch) QemuExecutable() string {
+func (arch Arch) QemuExecutable(kvmEnabled bool) string {
 	binaries := []string{
 		"qemu-system-" + arch.QemuArch(),
-		"qemu-kvm",
-		"/usr/libexec/qemu-kvm",
+	}
+	if kvmEnabled {
+		binaries = append(binaries,
```

Comment:
I'm confused a bit, this just changes the search path by removing `qemu-kvm` ? What does it do in terms of features?

Reply:
I believe the correct way to handle this parameter would be the function at the bottom of this file: https://github.com/siderolabs/talos/blob/27d771e0fe355f54079bde5566c56143b027431a/pkg/provision/providers/qemu/arch.go#L243

This is because we should first exclude the argument demanding the use of KVM rather than finding a binary that does not include the KVM acceleration support. However `qemu-kvm` command might force KVM enabled on some distros: not really sure about that

Reply:
I haven't tested that myself, but I guess `kvm` is enabled by default, so we can skip forcing it (which would fix the case when KVM is not available), but without KVM emulation would be super slow

Reply:
> I'm confused a bit, this just changes the search path by removing qemu-kvm

Yes, this line just removes `qemu-kvm`binary  from the search list if KVM is disabled. Maybe `qemu-kvm` can be used even without kvm, but I think it'd be confusing and I haven't looked into it.

> What does it do in terms of features?

Do you mean the PR in general? It allows to disable KVM usage, which was mandatory before and would error if KVM could not be used. Now a user can optionally set `--with-kvm` to false and thus run talos via qemu even on a system which doesn't have kvm (e.q. wsl, docker on a non linux host or another vm).

Reply:
I think `qemu-kvm` is just yet another alias present on some systems, and removing it from the search path doesn't make any difference, or make it more confusing in the end. 

`talosctl cluster create` doesn't work on non-Linux, as CNI binaries used to set up networking won't work on non-Linux hosts, so the only case is Linux with KVM disabled, which probably exists, but rare to find.

Reply:
Would it then make more sense to just log a warning and proceed with KVM off when it's not available?
I think it still makes sense as it's not a big change and would aid in development and testing of the create command. At least I would benefit from it (see pr description)